### PR TITLE
Fix issue #426 on plugins/tga.py

### PR DESCRIPTION
**Pull request by fozzysec**
Fix validation schema for longzhu.com in plugins/tga.py and standardize the returned stream name.

### CODE REVIEW

**Comment by beardypig** [comment]
Thanks for the fix. 

I just have a couple of notes about the stream names. The streams should be named for their quality not the streaming technology. The "live" streams should be called "live" (unless the quality is known) and the other streams should be the quality name.

If two streams have the same name then they are given as alternative streams in the UI. If a user has specified they don't want to use a particular streaming technology those are filtered out. Additionally, the "best" and "worst" streams cannot be determined with the appropriate names.

**Comment by beardypig** [comment]
@fozzysec you should rebase instead of merge if you want to update your branch.

**Comment by steven7851** [comment]
Use new api to get high quality stream url, old api is only 720p max.
```python
PLU_STREAM_INFO_URL = "http://livestream.plu.cn/live/getlivePlayurl?roomId=%d"
```

**Comment by fozzysec** [comment]
@steven7851 I found some problem with the new API, even though the new API returned the stream as 1080p, the actually output stream is not 1080p, it is 720p actually.
`{"liveUrl":"http://star.longzhu.com/117984/","inbandwidth":1566.29296875,"isTransfer":0,"playLines":[{"lineType":0,"urls":[{"rateLevel":3,"securityUrl":"http://flvtx.plu.cn/onlive/ce9d0e58701243d4b7dd2441d6664356.flv?txSecret=ee6c06786c1090bc32ec3fd5bd3abc46&txTime=588bb16c","resolution":"1920x1080","timeMove":false,"ext":"flv"},{"rateLevel":3,"securityUrl":"rtmp://rtmptx.plu.cn/onlive/ce9d0e58701243d4b7dd2441d6664356?txSecret=ee6c06786c1090bc32ec3fd5bd3abc46&txTime=588bb16c","resolution":"1920x1080","timeMove":false,"ext":"rtmp"},{"rateLevel":3,"securityUrl":"http://hlstx.plu.cn/onlive/ce9d0e58701243d4b7dd2441d6664356.m3u8?txSecret=be04d8e0291df66272d2fd45525f21b8&txTime=588d0094","resolution":"1920x1080","timeMove":false,"ext":"m3u8"}],"playLiveStreamType":12}],"pushLiveStreamType":12,"liveSourceType":8,"defaultLine":0,"defaultRateLevel":3}`
This is the new API returned json, and the stream is actually 720p.
`streamlink http://star.longzhu.com/117984\?from\=challcontent 1080p_http -O |avconv -i pipe:0
avconv version 11.4, Copyright (c) 2000-2014 the Libav developers
  built on Sep 13 2016 05:26:47 with Apple LLVM version 8.0.0 (clang-800.0.38)
[cli][info] Found matching plugin tga for URL http://star.longzhu.com/117984?from=challcontent
[cli][info] Available streams: 1080p_http, 1080p_hls, 1080p (best, worst)
[cli][info] Opening stream: 1080p_http (http)
[flv @ 0x7f97f780b000] max_analyze_duration 5000000 reached
Input #0, flv, from 'pipe:0':
  Metadata:
    fileSize        : 0
    audiochannels   : 2
    encoder         : Open Broadcaster Software v0.659b
  Duration: 00:00:00.00, start: 31053.768000, bitrate: N/A
    Stream #0.0: Video: h264 (High), yuv420p, 1280x720, 1536 kb/s, 25 fps, 1k tbn, 50 tbc
    Stream #0.1: Audio: aac, 48000 Hz, stereo, fltp, 131 kb/s
At least one output file must be specified
[cli][error] Error when writing to output: [Errno 32] Broken pipe
[cli][info] Stream ended
[cli][info] Closing currently open stream...
`
Besides, I do not understand the validation usage, I wrote something like
`{"playLines": [{ "urls": [{ ...}] {]}` but it doesn't work, would you please help me correct the validator?

**Comment by steven7851** [comment]
```python
_plu_schema = validate.Schema(
    {
        "playLines": [{
            "urls": [{
                "securityUrl": validate.url(scheme=validate.any("rtmp", "http")),
                "resolution": validate.text,
                "ext": validate.text
            }]
        }]
    }
)
```
```python
for source in info["playLines"][0]["urls"]:
```

**Comment by fozzysec** [comment]
@steven7851 I got it, I forgot to change info and will get it fixed soon. Thanks a lot.

File: src/streamlink/plugins/tga.py [hunk 1]
```diff
@@ -82,16 +80,16 @@ def _get_channel_id(self, domain):
     def _get_qq_streams(self, vid):
         res = http.get(QQ_STREAM_INFO_URL % (vid, 1))
         info = http.json(res, schema=_qq_schema)
-        yield "live", HTTPStream(self.session, info)
+        yield "http", HTTPStream(self.session, info)
```
Comment (beardypig) [inline src/streamlink/plugins/tga.py]: This should remain `"live"`.

File: src/streamlink/plugins/tga.py [hunk 2]
```diff
@@ -82,16 +80,16 @@ def _get_channel_id(self, domain):
...
 
         res = http.get(QQ_STREAM_INFO_URL % (vid, 2))
         info = http.json(res, schema=_qq_schema)
-        yield "live_http", HLSStream(self.session, info)
+        yield "hls", HLSStream(self.session, info)
```
Comment (beardypig) [inline src/streamlink/plugins/tga.py]: This should also be `"live"`.

**Comment by gravyboat** [comment]
@beardypig I think we're good to go here yeah? We have a few PRs piling up and I would like to get them in so people are properly attributed for their fixes!

File: docs/plugin_matrix.rst [hunk 3]
```diff
@@ -139,6 +139,7 @@ tf1                 - tf1.fr             Yes   No    Streams may be geo-restrict
                     - lci.fr
 tga                 - star.plu.cn        Yes   No
                     - star.tga.plu.cn
+		    - star.longzhu.com
```
Comment (bastimeyer) [inline docs/plugin_matrix.rst]: The indentation is wrong here... Please use spaces instead of tabs

**Comment by gravyboat** [comment]
Going to merge this in since everyone seems happy with the changes. Thanks for your contribution @fozzysec and sorry about the delay in getting this merged!

**Pull request merged**